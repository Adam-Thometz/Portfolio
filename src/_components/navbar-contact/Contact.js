import React from "react";
import { Link } from "react-router-dom";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faEnvelope } from "@fortawesome/free-solid-svg-icons";
import { faLinkedin, faGithub, faSoundcloud } from '@fortawesome/free-brands-svg-icons'
import './Contact.css';

const Contact = () => {
  const openEmail = () => window.open("mailto:adamthometz@gmail.com");
  const openLinkedIn = () => window.open("https://www.linkedin.com/in/adam-thometz/");
  const openGithub = () => window.open("https://github.com/Adam-Thometz");
  const openSoundcloud = () => window.open("https://soundcloud.com/this-is-psychic-hamster");

  return (
    <div className="Contact">
      <Link className="Contact-link email" to="#" onClick={openEmail}><FontAwesomeIcon title="Email" icon={faEnvelope} /></Link>
      <Link className="Contact-link linkedin" to="#" onClick={openLinkedIn}><FontAwesomeIcon title="LinkedIn" icon={faLinkedin} /></Link>
      <Link className="Contact-link github" to="#" onClick={openGithub}><FontAwesomeIcon title="Github" icon={faGithub} /></Link>
      <Link className="Contact-link soundcloud" to="#" onClick={openSoundcloud}><FontAwesomeIcon title="Soundcloud" icon={faSoundcloud} /></Link>
    </div>
  )
};

export default Contact;