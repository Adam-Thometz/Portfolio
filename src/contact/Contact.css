.Contact {
  padding: 30px 0;
  border-radius: 50px;
  background-image: linear-gradient(rgb(24, 63, 92), rgb(95, 40, 40));
  height: 80%;
  box-shadow: 5px 10px 20px 20px rgb(24, 63, 92);
  overflow: scroll; 
}

.Contact-form {
  padding: 0 30%;
}

.Contact-user-info {
  display: flex;
}

.Contact h2,.Contact p {
  text-align: left;
  margin: 12px 0;
}

.Contact-submit {
  border: 1px solid #171717;
  border-radius: 10px;
  padding: 8px 24px;
  background-color: #00ACCA;
  color: #8D1C1C;
  font-size: 1.5rem;
  font-family: 'Nunito', sans-serif;
  cursor: pointer;
  transition: background-color .5s ease-in-out;
}

.Contact-submit:hover {
  background-color: #c38585;
}

@media (max-width: 700px) {
  .Contact {
    padding: 30px 10%;
  }
}

@media (max-width: 500px) {
  .Contact-user-info {
    flex-direction: column;
  }  
}