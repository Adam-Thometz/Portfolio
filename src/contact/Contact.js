import React, { useState, useRef } from 'react';

import './Contact.css';

import Input from '../_components/input/Input';

import emailjs from '@emailjs/browser';
import { userId, serviceId, templateId } from '../_data/emailInfo';

const Contact = () => {
  const [formData, setFormData] = useState({
    name: '',
    email: '',
    message: ''
  });
  const form = useRef();
  const submit = useRef();
  
  const handleChange = e => {
    const { name, value } = e.target;
    setFormData(data => ({ ...data, [name]: value }));
  };

  const sendEmail = e => {
    e.preventDefault();
    submit.current.value = "Sending..."
    emailjs.sendForm(serviceId, templateId, form.current, userId)
    .then(() => {
      alert('Thank you for contacting me! I will get back to you when I can.');
      submit.current.value = "Submit";
    }, () => {
      alert('Something went wrong. Please try again later.');
      submit.current.value = "Submit";
    });
  };

  return (
    <form ref={form} className='Contact' onSubmit={sendEmail}>
      <h2>Send me a message!</h2>
      <p>Tell me anything! A joke, a Switch game I should try, or your open positions :)</p>
      <section className='Contact-user-info'>
        <Input label="Name" name="name" id="name" value={formData.name} onChange={handleChange} />
        <Input label="Email" name="email" id="email" value={formData.email} onChange={handleChange} type='email' />
      </section>
      <Input label="Message" name="message" id="message" value={formData.message} onChange={handleChange} type='textarea' />
      <input ref={submit} className='Contact-submit' type='submit' value='Submit' />
    </form>
  );
};

export default Contact;