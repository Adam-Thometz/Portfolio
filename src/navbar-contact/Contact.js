import React from "react";
import { Link } from "react-router-dom";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faEnvelope } from "@fortawesome/free-solid-svg-icons";
import { faLinkedin, faGithub } from '@fortawesome/free-brands-svg-icons'
import './Contact.css';

const Contact = () => {
  return (
    <div className="Contact" tabIndex={0}>
      <Link className="Contact-link email" to="#" onClick={() => window.open("mailto:adamthometz@gmail.com")}><FontAwesomeIcon title="Email" icon={faEnvelope} /></Link>
      <Link className="Contact-link linkedin" to="#" onClick={() => window.open("https://www.linkedin.com/in/adam-thometz/")}><FontAwesomeIcon title="LinkedIn" icon={faLinkedin} /></Link>
      <Link className="Contact-link github" to="#" onClick={() => window.open("https://github.com/Adam-Thometz")}><FontAwesomeIcon title="Github" icon={faGithub} /></Link>
    </div>
  )
};

export default Contact;