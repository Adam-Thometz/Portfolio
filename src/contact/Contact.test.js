import React from "react";
import { render } from "@testing-library/react";
import Contact from "./Contact";

describe("Contact component", () => {
  it("renders without crashing", () => {
    render(<Contact />);
  });

  it("matches the snapshot", () => {
    const { asFragment } = render(<Contact />);
    expect(asFragment()).toMatchSnapshot();
  });
});
