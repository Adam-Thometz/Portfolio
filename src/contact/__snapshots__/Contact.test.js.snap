// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact component matches the snapshot 1`] = `
<DocumentFragment>
  <form
    class="Contact"
  >
    <h2>
      Let's start a dialogue! 👋🏼
    </h2>
    <section
      class="Contact-user-info"
    >
      <fieldset
        class="Input"
      >
        <input
          id="name"
          name="name"
          required=""
          type="text"
          value=""
        />
        <label
          for="name"
        >
          Name
        </label>
      </fieldset>
      <fieldset
        class="Input"
      >
        <input
          id="email"
          name="email"
          required=""
          type="email"
          value=""
        />
        <label
          for="email"
        >
          Email
        </label>
      </fieldset>
    </section>
    <fieldset
      class="Input"
    >
      <textarea
        id="message"
        name="message"
        required=""
      />
      <label
        for="message"
      >
        Message
      </label>
    </fieldset>
    <input
      class="Contact-submit"
      type="submit"
      value="Submit"
    />
  </form>
</DocumentFragment>
`;
